ui/ui.gradle.kts

```diff
@@ -0,0 +1,195 @@
+            }
+            exec {
+                workingDir = pipelibDir
+                commandLine = listOf("C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\vcvarsall.bat", "amd64", "&", "cmake", "--build", ".", "--target", "ALL_BUILD", "--config", "Release")
+            }
+            exec {
+                workingDir = testClassesDir
+                commandLine = listOf("cmd", "/C", "copy /Y pipelib\\Release\\* .")
+            }
+            exec {
+                workingDir = testClassesDir
+                commandLine = listOf("cmd", "/C", "copy /Y pipelib\\pipe\\Release\\* .")
+            }
+            exec {
+                workingDir = pipelibDir
+                commandLine = listOf("cmd", "/C", "copy /Y ..\\..\\..\\..\\src\\test\\resources\\edu\\wpi\\grip\\ui\\codegeneration\\tools\\pipe\\AbsPipeline.h pipe\\")
+            }
+            exec {
+                workingDir = testClassesDir
+                commandLine = listOf("cmd", "/C", "copy /Y ..\\..\\..\\src\\test\\resources\\edu\\wpi\\grip\\ui\\codegeneration\\tools\\realpipe\\CMakeLists.txt .")
+            }
+        } else {
+            exec {
+                isIgnoreExitValue = true // if clean hasn"t been called, directory already exists and mkdir fails.
+                workingDir = testClassesDir
+                commandLine = listOf("mkdir", "pipelib")
+            }
+            exec {
+                workingDir = pipelibDir
+                commandLine = listOf("cmake", "../../../../src/test/resources/edu/wpi/grip/ui/codegeneration/tools/")
+            }
+            exec {
+                workingDir = pipelibDir
+                commandLine = listOf("make")
+            }
+            exec {
+                workingDir = testClassesDir
+                if (syst == "osx") {
+                    commandLine = listOf("cp", "pipelib/libgenJNI.dylib", ".")
+                }
+                if (syst == "linux") {
+                    commandLine = listOf("cp", "pipelib/libgenJNI.so", ".")
+                }
+            }
+            exec {
+                workingDir = pipelibDir
+                commandLine = listOf("cp", "../../../../src/test/resources/edu/wpi/grip/ui/codegeneration/tools/pipe/AbsPipeline.h", "pipe/")
+            }
+            exec {
+                workingDir = testClassesDir
+                commandLine = listOf("cp", "../../../src/test/resources/edu/wpi/grip/ui/codegeneration/tools/realpipe/CMakeLists.txt", ".")
+            }
+        }
+    }
+}
+
+if (project.hasProperty("generation") || project.hasProperty("genonly")) {
+    val syst = osdetector.os
+    tasks.withType<Test>() {
+        val platform = Installer.getPlatform()
+        Installer.setOpenCvVersion("3.1.0")
+        val jniLocation = project.properties.getOrDefault("jniLocation", platform.defaultJniLocation())
+        Installer.installJni("$jniLocation")
+        val defaultLibPath = System.getProperty("java.library.path");
```

Comment:
Can this installation stuff happen in a task instead of in the Gradle configuration phase?

Reply:
Done